src/__tests__/http-test.ts

```diff
@@ -860,7 +859,7 @@ function runTests(server: Server) {
         .set('Content-Type', 'application/json')
         .set('Content-Encoding', 'gzip');
 
-      req.write(zlib.gzipSync('{ "query": "{ test }" }'));
+      await req.write(gzipSync('{ "query": "{ test }" }'));
```

Comment:
`write` is typed as returning a Promise

Reply:
Can you please extract it into separate PR and fix it still in Flow?
Also can you please reference the `write` definition so I can see it?

Reply:
@IvanGoncharov I guess this is more of a false positive.

<img width="647" alt="Screen Shot 2020-06-17 at 3 46 51 PM" src="https://user-images.githubusercontent.com/18018864/84943440-5716ca00-b0b2-11ea-8755-e877d3eb788f.png">

The result of calling `write` is a thenable and is not assigned to a variable so ESLint thinks there's a floating Promise.

Other than awaiting the code, we could also either ignore the `no-floating-promises` rule or chain the method with the preceding ones:

```
const req = app
        .request()
        .post(urlString())
        .set('Content-Type', 'application/json')
        .set('Content-Encoding', 'gzip')
        .write(gzipSync('{ "query": "{ test }" }'));
```

Reply:
I think adding the method to the existing chain makes the most sense. Off-hand, I'm not sure why that wasn't done like that in the first place.

Reply:
> I think adding the method to the existing chain makes the most sense. Off-hand, I'm not sure why that wasn't done like that in the first place.

Agree, make total sense 👍

Reply:
@IvanGoncharov So apparently the reason we wrote the tests this way to begin with is because [write returns a boolean instead of `this`](https://github.com/visionmedia/supertest/issues/433). So it can't be chained like the other methods. This also means that `@types/superagent` is wrong :(

I could just add the `await` like I originally did and that would resolve the linter issue. Do you have a preference for how we address this?

Reply:
@IvanGoncharov ping

Reply:
@danielrearden Can you open PR against `supertest` to fix this?
I tried to look for a better library but it sims there nothing better ATM.

Since it's only a few places like that I'm totally ok with adding eslint-disable comments and reference supertest's PR.